### ISSUE TITLE
Find refs/rename not working for mapped types

### ISSUE THREAD

**Issue by OliverJAsh** [Microsoft/TypeScript#18548]
v2.5.2

``` ts
{
    abstract class Bar<Props> {
        props: Readonly<Props>;
    }
    type Props = { foo: string };
    class Foo extends Bar<Props> {
        constructor() {
            super()
            const { foo } = this.props;

            foo
        }
    }
}
```

As you can see from the following screenshot, the `foo` property in the `Props` type is not highlighted, and thus not found/used by "find all references" or "rename".

![image](https://user-images.githubusercontent.com/921609/30551510-93bc4a5c-9c92-11e7-8723-ab7d584edf1a.png)

This seems very related to this bug which was apparently fixed in v2.4. The test case mentioned there does work, so perhaps this is a different bug https://github.com/Microsoft/TypeScript/issues/13690#issuecomment-277244978

This example is identical to React's typings for components. Currently we can't use references/rename on our props :-(

**Comment by mhegazy** [comment]
seems working in latest.
![image](https://user-images.githubusercontent.com/8000722/30553103-f5133060-9c54-11e7-8509-02b6e10a8901.png)

**Comment by OliverJAsh** [comment]
@mhegazy I just tested it with `@next` and it works, but it fails with `@insiders` and `@latest` for me. Are you seeing different results?

My VSCode is definitely set to use the correct TS version. See bottom right

![image](https://user-images.githubusercontent.com/921609/30554180-864de980-9c9b-11e7-9397-e5e0d24a60ad.png)

**Comment by OliverJAsh** [comment]
If above is correct, we can probably assume this will be fixed in 2.6?

**Comment by mhegazy** [comment]
VSCode ships with last stable (2.5.2), insiders ships with latest `@insiders` (what is going to be the 2.5.3 release). 

You can use `@next`, see [Using Newer TypeScript Versions](https://code.visualstudio.com/docs/languages/typescript#_using-newer-typescript-versions) documentation for more details on updating your VSCode to use a different version of TypeScript.

**Comment by mhegazy** [comment]
> If above is correct, we can probably assume this will be fixed in 2.6?

That is correct.

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
